base/logging.go

```diff
@@ -409,10 +409,10 @@ func Tracef(logKey LogKey, format string, args ...interface{}) {
 
 func logTo(logLevel LogLevel, logKey LogKey, format string, args ...interface{}) {
 	shouldLogConsole := consoleLogger.shouldLog(logLevel, logKey)
-	shouldLogError := errorLogger.shouldLog()
-	shouldLogWarn := warnLogger.shouldLog()
-	shouldLogInfo := infoLogger.shouldLog()
-	shouldLogDebug := debugLogger.shouldLog()
+	shouldLogError := errorLogger.shouldLog(logLevel)
+	shouldLogWarn := warnLogger.shouldLog(logLevel)
+	shouldLogInfo := infoLogger.shouldLog(logLevel)
+	shouldLogDebug := debugLogger.shouldLog(logLevel)
```

Comment:
Is the logLevel check below as part of the early return (line 420) still needed?  If so, should we move the 'logLevel <= LevelNone' check to the top of the function (before line 411) to *early* early return?

Reply:
That check is more of a defensive measure than functional, all callers to `logTo` (`Debugf`, `Infof`, `Warnf`, etc.) supply a logLevel that meets that criteria.

I suppose it could be moved right up to the top though!